includes/cli/class-woocommerce-subscriptions.php

```diff
@@ -228,11 +245,32 @@ private static function get_subscriptions( $page = 1 ) {
 			$subscriptions = wcs_get_subscriptions(
 				[
 					'paged'                  => $page,
-					'subscriptions_per_page' => 25,
+					'subscriptions_per_page' => 50,
 					'subscription_status'    => 'on-hold',
 				]
 			);
 		}
 		return $subscriptions;
 	}
+
+	/**
+	 * Get flagged subscriptions to update.
+	 *
+	 * @return array
+	 */
+	private static function get_flagged_subscriptions() {
+		$subscriptions = wcs_get_subscriptions(
+			[
+				'subscriptions_per_page' => 50,
+				'subscription_status'    => 'on-hold',
+				'meta_query'             => [ // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+					[
+						'key'     => '_newspack_cli_status_updated',
```

Comment:
Suggestion: Use a different meta key to mark subscriptions to be processed at the end of the run. Otherwise you might end up processing subscriptions marked in a previous run to the script.

then you should delete this meta after the sub is processed...

or... just store the subs that need to be processed in the memory. You can store only the ID and get the subscription at the end...

Makes sense?

Reply:
In any case, I updated to use the `_newspack_cli_to_status` meta as the flag in https://github.com/Automattic/newspack-plugin/pull/3750/commits/9021ec6b9dde6e5ceaefd6294e1802794a07a460